### PR TITLE
Feature/map publisher node

### LINKED ISSUE #22: Using RViz as simulation visualisation
Current TODO

- [x] Determine exact scope of what we want to visualize
    - [ ] Determine how we want to show collisions 
    - [x] Publish the map, or  the mesh that is generated internally?
  
- [x] First proof of concept tf2-publishers
    - [ ] Configure the static base_link->LRF transformation, such that they match Hero's

- [x] Clearly Indicate the current Robot position
     - [x]  Currently only tf-origin is shown

- [x] Launch file and rviz-config file

- [x] Verify that it plays nice together with other viz markers

### PR DESCRIPTION
**Pull request by KdVos**
As stated in #26, based on the work by @ThijsBeurskensTUE, a restructured version to achieve the same set of features.
New pull request due to the large structural differences compared to the first version

Closes #22

### CODE REVIEW

File: src/visualization.cpp [hunk 1]
```diff
@@ -191,4 +191,93 @@ void visualize(const World& world, const std::vector<RobotPtr>& robots, bool col
     cv::waitKey(3);
 }
 
+visualization_msgs::MarkerArray create_rviz_objectmsg(const World &world, const MapConfig &mapconfig)
```
Comment (PetervDooren) [inline src/visualization.cpp:194]: This function visualises the map as well as the objects. question is whether this is how we want to do visualisation. As it means we cannot visualise on the robot the same way we do in simulator

**Review by PetervDooren — CHANGES_REQUESTED** [review]

File: src/visualization.cpp [hunk 2]
```diff
@@ -191,4 +191,93 @@ void visualize(const World& world, const std::vector<RobotPtr>& robots, bool col
...
+            object.color.r = 1.0;
+            object.color.b = 0.0;
+            object.color.g = 0.0;
+        }
+        else
+        {
+            object.color.r = 0.0;
+            object.color.b = 0.0;
+            object.color.g = 0.0;
+        }
```
Comment (KdVos) [inline src/visualization.cpp:250]: change here probably

File: src/visualization.cpp (hunk 1 above)
Comment (KdVos) [reply src/visualization.cpp:194]: see below

Comment (KdVos) [reply src/visualization.cpp:194]: robots and walls are skipped in lines 229 and 231

File: src/robot.cpp [hunk 3]
```diff
@@ -18,6 +18,26 @@ void Robot::speakCallback(const std_msgs::String::ConstPtr& msg)
...
+
+    tf2::Matrix3x3 T(q);
+
+    double roll, pitch, yaw;
+    T.getRPY(roll, pitch, yaw);
+
+    mapconfig.mapOrientation = yaw + M_PI/2;
+    mapconfig.mapInitialised = true;
+}
+
```
[the remark this answers is not in the document]
Comment (KdVos) [reply src/robot.cpp:40]: true

File: src/main.cpp [hunk 4]
```diff
@@ -293,7 +293,10 @@ int main(int argc, char **argv){
             // Write tf2 data
             geo::Pose3D pose = world.object(robot.robot_id).pose;
             if (robot.mapconfig.mapInitialised)
+            {
+                robot.sub_mapdata.shutdown();
```
Comment (PetervDooren) [inline src/main.cpp]: I thinnk you can place this at the end of MapCallback.

**Review by PetervDooren — APPROVED** [review]

**Pull request merged**